hack/tools.mk

```diff
@@ -175,6 +176,9 @@ $(OPENAPI_GEN): go.mod
 $(PROMTOOL): $(TOOLS_PKG_PATH)/install-promtool.sh
 	@$(TOOLS_PKG_PATH)/install-promtool.sh
 
+$(PROTOC):
+	@$(TOOLS_BIN_DIR)/../../download-protoc.sh
```

Comment:
The version should be defined in a make variable (like the others) and passed as an argument to the script.
This way, dependencies that reuse this makefile can overwrite the version if complications arise.

Reply:
Also, we should add a version file so that make installs a new version of `protoc` if we upgrade it.

Reply:
I have added the version dependency for `protoc` in 5aaa1cde6a18adbeec5834d356386c86320cb63e, but this is causing the tool to be built every time, I am not able to figure out why is this the case. PTAL.

Reply:
This is due to the version file being newer than the binary itself:
```
$ make -d hack/tools/bin/protoc
...
Finished prerequisites of target file 'hack/tools/bin/protoc'.
 Prerequisite 'hack/tools/bin/.version_protoc_23.4' is newer than target 'hack/tools/bin/protoc'.
Must remake target 'hack/tools/bin/protoc'.
hack/tools.mk:181: update target 'hack/tools/bin/protoc' due to: hack/tools/bin/.version_protoc_23.4
```

This in turn happens because unzip preserves timestamps. I.e., the modification timestamp of the binary will be the release build date:
```
$ stat hack/tools/bin/protoc
  File: hack/tools/bin/protoc
  Size: 12317760  	Blocks: 24064      IO Block: 4096   regular file
Device: 1,16	Inode: 21145698    Links: 1
Access: (0555/-r-xr-xr-x)  Uid: (  503/ ebertti)   Gid: (   20/   staff)
Access: 2023-07-11 08:27:35.955428061 +0200
Modify: 2023-07-06 00:39:05.000000000 +0200
Change: 2023-07-11 08:27:35.879341030 +0200
 Birth: 2023-07-06 00:39:05.000000000 +0200
```

You should be able to resolve this by replacing `mv` with `cp`:
```diff
- mv -f "$out_dir"/bin/protoc "$TOOLS_BIN_DIR"/protoc
+ cp -f "$out_dir"/bin/protoc "$TOOLS_BIN_DIR"/protoc
```

Reply:
Thank you 👍